relayer/chains/cosmos/tx.go

```diff
@@ -246,13 +262,7 @@ func (cc *CosmosProvider) buildMessages(ctx context.Context, msgs []provider.Rel
 		}
 		return nil
 	}, retry.Context(ctx), rtyAtt, rtyDel, rtyErr); err != nil {
-		return nil, sdk.Coins{}, err
-	}
-
-	// Attach the signature to the transaction
-	// Force encoding in the chain specific address
-	for _, msg := range msgs {
-		cc.Codec.Marshaler.MustMarshalJSON(CosmosMsg(msg))
+		return nil, 0, sdk.Coins{}, err
 	}
```

Comment:
It looks like this loop and the logic associated with it are gone in your proposed changes. I see we build the unsigned tx but where do we now attach the signature? Perhaps this was unnecessary and the tx is signed elsewhere but figured I would point this out.

Reply:
The return value of these `MustMarshalJSON` calles were ignored, so this code wasn't doing anything.